Updated release notes and README to prepare for 3.0a3 (#107)
modelon-community/fmi-library: RELEASE_NOTES.md

Note that version 2.1 is the first version with release notes. Please see the commit history for older versions.

## 3.0a3

### Improvements

- Improved error handling.

> - Attribute buffers are now properly cleared if an elements fail to parse.

Reviewer: ```suggestion
- Attribute buffers are now properly cleared if an element fails to parse.
```

> - Disabled jm_locale_test if locale testing is disabled.
> ### Changes

Reviewer: ```suggestion
- Disabled jm_locale_test if locale testing is disabled.

### Changes
```

> - Resolved reported vulnerability issues by:
>     - Updated to expat 2.5.0 up from version 2.8.0.
>     - Updated to zlib 1.2.13 up from version 1.2.6.

Reviewer: ```suggestion
- Resolved reported vulnerability issues:
    - Updated to expat 2.5.0 (from version 2.8.0).
    - Updated to zlib 1.2.13 (from version 1.2.6).
```

Reviewer 2: ```
- Updated to expat 2.5.0 (from version 2.4.8). ?
```

Author: Good catch, thank you!

> - Renamed `default_experiment_step` to `_step_size`.

Reviewer: This doesn't make any sense to me.
